physionet-django/project/authorization/access.py

```diff
@@ -0,0 +1,100 @@
+from datetime import datetime
+
+from django.db.models import Q
+
+from events.models import Event, EventDataset
+from project.models import AccessPolicy, DUASignature, DataAccessRequest, PublishedProject
+from user.models import Training, TrainingType
+
+
+def get_accessible_projects(user):
```

Comment:
The logic in this function is difficult to follow. I wonder whether splitting it into some subfunctions might help with readability?

Reply:
Thanks @tompollard  working on this. Also looks like another change by Rafal was merged which kinda of make the new logic a bit difficult to rebase quickly, i will edit the last commit on this PR and make changes.